x/ecocredit/base/keeper/msg_create_project.go

```diff
@@ -41,14 +45,26 @@ func (k Keeper) CreateProject(ctx context.Context, req *types.MsgCreateProject)
 		return nil, err
 	}
 
-	if err = k.stateStore.ProjectTable().Insert(ctx, &api.Project{
-		Id:           projectID,
-		Admin:        adminAddress,
-		ClassKey:     classInfo.Key,
-		Jurisdiction: req.Jurisdiction,
-		Metadata:     req.Metadata,
-		ReferenceId:  req.ReferenceId,
-	}); err != nil {
+	project.Admin = adminAddress
+	project.Jurisdiction = req.Jurisdiction
+	project.Metadata = req.Metadata
+	if req.ReferenceId != "" {
+		project.ReferenceId = req.ReferenceId
+		// only set deprecated class key if reference id is not empty to support the use case of reference IDs
+		project.ClassKey = classInfo.Key //nolint:staticcheck
+	}
```

Comment:
It seems that the reference ID can only be set by credit class issuers when directly creating + enrolling a project into a credit class via `MsgCreateProject` and there is no way to update this value. If a project is created by an admin and applies to a credit class it doesn't seem that the reference ID could be set. Is this intentional? Would it make sense to move the reference ID to the ProjectEnrollmentTable?

Reply:
The reference ID is only relevant in the case of bridging and I can consider it a legacy use case. It needs to be in the project table to impose an unique constraint on it, again only needed for bridging.

Reply:
> The reference ID is only relevant in the case of bridging and I can consider it a legacy use case. 

Oh interesting, I did not realize this was only intended for bridging. In fact we are using the reference ID field in another way on some projects we are working on (SeaTrees, ZFP). It is useful as a key to reference on-chain projects with existing "external IDs" programs may already be using. This makes it much easier to create scripts that update projects in bulk when metadata might be changing in an external CMS of some kind.

It would be good to know if this will be supported going forward. These projects/processes are being deployed on Redwood now and could be deployed on mainnet in the coming months.

> // reference_id is any arbitrary string that can be use to reference project.

https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.v1#regen.ecocredit.v1.ProjectInfo

Reply:
I don't think we're planning to remove the support, but I'd prefer not to add or update the functionality much unless there's a specific use case. Basically, it can only be added at creation, not modified and there will be an unique constraint on reference ID and credit class.